### PR TITLE
PMM-825: run `pt-(*-)summary` tool on given instance

### PR DESCRIPTION
**Pull request by arvenil**
PMM-825: run `pt-(*-)summary` tool on given instance
(https://github.com/percona/qan-api/pull/28)
```diff
 + `query`
     + Explain (EXPLAIN a query)
     + TableInfo (get table def, status, and indexes)
+    + Summary (run `pt-(*-)summary` tool if available) 
 ```

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -493,10 +493,13 @@ func (agent *Agent) Handle(cmd *proto.Cmd) *proto.Reply {
 		*/
 		agent.client.Disconnect()
 		return nil // no reply
+	// @obsolete by query/plugin/os/summary
```
Comment (AlekSi) [inline agent/agent.go]: What tool do you use for that `@obsolete` "annotation"?

File: pct/cmd/cmd.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func runCmd(cmd *exec.Cmd) (resultChan chan result) {
 	// because we might get data before we would be waiting on this channel
 	resultChan = make(chan result, 1)
 	go func() {
-		output, err := cmd.Output()
+		output, err := cmd.CombinedOutput()
 		select {
 		case resultChan <- result{output: string(output), err: err}:
 		default:
```
Comment (AlekSi) [inline pct/cmd/cmd.go:118]: There is no reason to use `select`: `resultChan` is buffered, first `case` always work.

File: agent/agent.go (hunk 1 above)
Comment (arvenil) [reply agent/agent.go]: It's just my label, we don't use any tools for that (we don't even use go doc, yet)

Comment (AlekSi) [reply agent/agent.go]: Maybe we could use `TODO` for that? Collecting those markers is supported by default in many editors and IDEs.

File: pct/cmd/cmd.go (hunk 2 above)
Comment (arvenil) [reply pct/cmd/cmd.go:118]: It's an old code not changed in this PR, I can't touch everything in one PR

Comment (AlekSi) [reply pct/cmd/cmd.go:118]: Sure, but that's a sweet spot refactoring, I would never miss that one. 😄

Comment (arvenil) [reply pct/cmd/cmd.go:118]: Feel free to make PRs with fixes to `qan-agent`, there is a lot of things that needs to be fixed  so any PR is welcome.

File: agent/agent.go (hunk 1 above)
Comment (arvenil) [reply agent/agent.go]: done

**Comment by arvenil** [comment]
@rnovikovP will be ok to merge that for next release? This, with changes to `qan-app`, will fix the issue with not working `pt-mondodb-summary` (more details in PMM-825)

**Review by rnovikovP — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]
LGTM. :shipit:

**Pull request merged**